File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.schemas.user import UserPublic


class NotificationOut(BaseModel):
    id: uuid.UUID
    type: str
    entity_id: uuid.UUID | None
    is_read: bool
    created_at: datetime
    actor: UserPublic

    model_config = {"from_attributes": True}

File: frontend/src/types/notification.ts
import type { User } from "./user";

export interface Notification {
  id: string;
  type: "like" | "comment" | "connection_request" | "connection_accepted";
  entity_id: string | null;
  is_read: boolean;
  created_at: string;
  actor: User;
}
